src/main/java/io/supertokens/webserver/api/multitenancy/RemoveTenantAPI.java

```diff
@@ -45,20 +49,27 @@ public String getPath() {
     protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
         JsonObject input = InputParser.parseJsonObjectOrThrowError(req);
 
-        String tenantId = InputParser.parseStringOrThrowError(input, "tenantId", true);
-        if (tenantId != null) {
-            tenantId = tenantId.trim();
+        String tenantId = InputParser.parseStringOrThrowError(input, "tenantId", false);
+        if (tenantId.equals(TenantIdentifier.DEFAULT_TENANT_ID)) {
+            throw new ServletException(new BadPermissionException("Cannot delete public tenant, use remove app API instead"));
         }
 
         try {
             TenantIdentifier sourceTenantIdentifier = this.getTenantIdentifierWithStorageFromRequest(req);
 
-            super.handle(
-                    sourceTenantIdentifier,
-                    new TenantIdentifier(sourceTenantIdentifier.getConnectionUriDomain(), sourceTenantIdentifier.getAppId(), tenantId),
-                    resp);
```

Comment:
you need to make sure that the tenant is not deleting itself?

Reply:
Please clearly write down somewhere the exact permissions for each of the CRUD APIs for muti tenancy.. there still seems to be some misundersatnding